clientmanager/starter.py

```diff
@@ -2,19 +2,59 @@
 
 
 import argparse
+import dataclasses as dc
 import datetime as dt
 import getpass
 import os
 import time
 from pathlib import Path
 
+import boto3  # type: ignore[import]
 import htcondor  # type: ignore[import]
+import requests
 from rest_tools.client import RestClient
 from wipac_dev_tools import argparse_tools
 
 from . import condor_tools
 from .config import LOGGER
 
+boto3.set_stream_logger(name="botocore")
+
+
+@dc.dataclass
+class S3File:
+    """Wrap an S3 file."""
+
+    url: str
+    base_fname: str
+
+
+def s3ify(filepath: Path) -> S3File:
+    """Put the file in s3 and return info about it."""
+    s3_client = boto3.client(
+        "s3",
+        "us-east-1",
+        endpoint_url=os.getenv("EWMS_TMS_S3_URL"),
+        aws_access_key_id=os.getenv("EWMS_TMS_S3_ACCESS_KEY"),
+        aws_secret_access_key=os.getenv("EWMS_TMS_S3_SECRET_KEY"),
+    )
+
+    # POST
+    upload_details = s3_client.generate_presigned_post(
+        "clientmanager",
+        os.getenv("SKYSCAN_SKYDRIVER_SCAN_ID"),
+    )
+    with open(filepath, "rb") as f:
+        response = requests.post(
+            upload_details["url"],
+            data=upload_details["fields"],
+            files={"file": (filepath.name, f)},  # maps filename to obj
+        )
+    print(f"Upload response: {response.status_code}")
+    print(str(response.content))
+
+    return S3File(upload_details["url"], filepath.name)
```

Comment:
One note: you've used the `presigned_post` url here, which is correct for uploading.  But you need the presigned get url for downloading in the clients.  You probably want to generate that here and add it to S3File.  When generating the get url, you probably also want to set the expiration time (the default is 1 hour).